eplmp-server/eplmp-server-rest/src/main/java/org/polarsys/eplmp/server/rest/file/DocumentBinaryResource.java

```diff
@@ -211,6 +212,16 @@ public Response downloadDocumentFile(
 
         InputStream binaryContentInputStream = null;
 
+        DocumentIteration lastIteration = documentService.getDocumentRevision(new DocumentIterationKey(workspaceId, documentId, version, iteration).getDocumentRevision()).getLastIteration();
+
+        Set<BinaryResource> binaryresources = lastIteration.getAttachedFiles();
+
+        boolean isCheckedOut = documentService.getDocumentRevision(new DocumentIterationKey(workspaceId, documentId, version, iteration).getDocumentRevision()).isCheckedOut();
+
+        boolean isWorkingCopy = binaryresources.contains(binaryResource) && isCheckedOut;
+
```

Comment:
Why pass through all the binaryresource? Should be better to just check if the iteration is the latest one